File: Exercises_C++/1828.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int qntd;
    cin >> qntd;
    string s, r;

    for (int c = 1; c <= qntd; ++c) {
        cin >> s >> r;
		
		if (s == r) {
			cout << "Caso #" << c << ": De novo!" << endl;
		}


        else if (s == "tesoura") {
            if (r == "papel" || r == "lagarto") {
                cout << "Caso #" << c << ": Bazinga!" << endl;
            } else {
                cout << "Caso #" << c << ": Raj trapaceou!" << endl;
            }
        } 

        else if (s == "papel") {
            if (r == "pedra" || r == "Spock") {
                cout << "Caso #" << c << ": Bazinga!" << endl;
            } else {
                cout << "Caso #" << c << ": Raj trapaceou!" << endl;
            }
        }

        else if (s == "pedra") {
            if (r == "lagarto" || r == "tesoura") {
                cout << "Caso #" << c << ": Bazinga!" << endl;
            } else {
                cout << "Caso #" << c << ": Raj trapaceou!" << endl;
            }
        }

        else if (s == "lagarto") {
            if (r == "papel" || r == "Spock") {
                cout << "Caso #" << c << ": Bazinga!" << endl;
            } else {
                cout << "Caso #" << c << ": Raj trapaceou!" << endl;
            }
        }

        else if (s == "Spock") {
            if (r == "tesoura" || r == "pedra") {
                cout << "Caso #" << c << ": Bazinga!" << endl;
            } else {
                cout << "Caso #" << c << ": Raj trapaceou!" << endl;
            }
        }
    }
    return 0;
}

File: Exercises_Python/1828.py


vezes = int(input())

for c in range(1,vezes+1):
    jogador_1,jogador_2 = map(str, input().split())
    if jogador_1 == jogador_2:
        print('Caso #{}: De novo!'.format(c))
    elif (jogador_1 == 'tesoura') and ((jogador_2 == 'lagarto') or (jogador_2 == 'papel')):
        print('Caso #{}: Bazinga!'.format(c))
    elif (jogador_1 == 'papel') and ((jogador_2 == 'pedra') or (jogador_2 == 'Spock')):
        print('Caso #{}: Bazinga!'.format(c))
    elif (jogador_1 == 'pedra') and ((jogador_2 == 'lagarto') or (jogador_2 == 'tesoura')):
        print('Caso #{}: Bazinga!'.format(c))
    elif (jogador_1 == 'lagarto') and ((jogador_2 == 'Spock') or (jogador_2 == 'papel')):
        print('Caso #{}: Bazinga!'.format(c))
    elif (jogador_1 == 'Spock') and ((jogador_2 == 'pedra') or (jogador_2 == 'tesoura')):
        print('Caso #{}: Bazinga!'.format(c))
    elif (jogador_2 == 'tesoura') and ((jogador_1 == 'lagarto') or (jogador_1 == 'papel')):
        print('Caso #{}: Raj trapaceou!'.format(c))
    elif (jogador_2 == 'papel') and ((jogador_1 == 'pedra') or (jogador_1 == 'Spock')):
        print('Caso #{}: Raj trapaceou!'.format(c))
    elif (jogador_2 == 'pedra') and ((jogador_1 == 'lagarto') or (jogador_1 == 'tesoura')):
        print('Caso #{}: Raj trapaceou!'.format(c))
    elif (jogador_2 == 'lagarto') and ((jogador_1 == 'Spock') or (jogador_1 == 'papel')):
        print('Caso #{}: Raj trapaceou!'.format(c))
    elif (jogador_2 == 'Spock') and ((jogador_1 == 'pedra') or (jogador_1 == 'tesoura')):
        print('Caso #{}: Raj trapaceou!'.format(c))

        
    
    
